File: k9mail/src/main/java/com/fsck/k9/account/OutlookWebViewClient.java
package com.fsck.k9.account;

import android.net.Uri;

import com.fsck.k9.activity.setup.AccountSetupPresenter;


public class OutlookWebViewClient extends OAuth2WebViewClient {
    public OutlookWebViewClient(AccountSetupPresenter presenter) {
        super(presenter);
    }

    @Override
    protected boolean arrivedAtRedirectUri(Uri uri) {
        return ("msal" + "a41aa976-c5ad-405f-a8e3-ed18c07bb13a").equals(uri.getScheme());
    }

    @Override
    protected boolean getOutOfDomain(Uri uri) {
        return !uri.getHost().contains("live.com"); // TODO: 8/18/17 how to improve it?
    }
}

File: mail/common/src/main/java/com/fsck/k9/mail/oauth/outlook/OutlookWebViewClient.kt
package com.fsck.k9.mail.oauth.outlook

import android.net.Uri
import com.fsck.k9.mail.oauth.authorizationserver.codegrantflow.OAuth2CodeGrantFlowManager
import com.fsck.k9.mail.oauth.authorizationserver.codegrantflow.OAuth2WebViewClient

class OutlookWebViewClient(email: String, codeGrantFlowManager: OAuth2CodeGrantFlowManager) :
    OAuth2WebViewClient(email, codeGrantFlowManager) {
    override fun arrivedAtRedirectUri(uri: Uri?): Boolean {
        return "msal" + OutlookAuthorizationServer.CLIENT_ID == uri?.scheme
    }
}
